doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,282 @@
+  {
+          char *sender = (char *)arg;
+
+          if (error_flags) {
+                  LOG_ERR("Sendig failed [%d]\nSender: %s\n", error_flags, sender);
+          }
+  }
+
+  int send_function(struct device *can_dev)
+  {
+          struct zcan_frame frame = {
+                  .id_type = CAN_EXTENDED_IDENTIFIER,
+                  .rtr = CAN_DATAFRAME,
+                  .ext_id = 0x1234567,
+                  .dlc = 2
+          };
+
+          frame.data = {1,2};
+          return can_send(can_dev, &frame, K_FOREVER, tx_irq_callback, "Sender 1");
+  }
+
+This example shows how to send a frame with extended identifier 0x1234567 and
+two bytes of data. The provided callback is called when the message is sent, or
+an error occurred. Passing `K_FOREVER` to the timeout causes the function to block
+until a transfer mailbox is assigned to the frame or an error occurred. It does
+not block until the message is sent like the example above.
+
+Receiving
+*********
+
+Frames are only received when they match a filter.
+The following code snippets show how to receive frames by attaching filters.
+
+.. code-block:: C
+
+  void rx_callback_function(struct zcan_frame *frame, void *arg)
+  {
+          ... do something with the frame ...
+  }
+
+
+Here we have an example for a receiving callback.
+It is used for `can_attach_isr` or `can_attach_workq`.
+The argument arg is passed when the filter is attached.
+
+.. code-block:: C
+
+  const struct zcan_filter my_filter = {
+          .id_type = CAN_STANDARD_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .std_id = 0x123,
+          .rtr_mask = 1,
+          .std_id_mask = CAN_STD_ID_MASK
+  };
+  int filter_id;
+  struct device *can_dev;
+
+  can_dev = device_get_binding("CAN_0");
+
+  filter_id = can_attach_isr(can_dev, rx_callback_function, callback_arg, &my_filter);
+  if (filter_id < 0) {
+    LOG_ERR("Unable to attach isr [%d]", filter_id);
+  }
+
+The snippet above shows how to attach a filter with an interrupt callback.
+It is the most efficient but also the most critical way to receive messages.
+The callback function is called from an interrupt context, which means that the
+callback function should be as short as possible and must not block.
+Attaching ISRs is not allowed from userspace context.
+
+The filter for this example is configured to match the identifier 0x123 exactly.
+
+.. code-block:: C
+
+  const struct zcan_filter my_filter = {
+          .id_type = CAN_STANDARD_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .std_id = 0x120,
+          .rtr_mask = 1,
+          .std_id_mask = 0x7F0
+  };
+  struct zcan_work rx_work;
+  int filter_id;
+  struct device *can_dev;
+
+  can_dev = device_get_binding("CAN_0");
+
+  filter_id = can_attach_workq(can_dev, &k_sys_work_q, &rx_work, callback_arg, callback_arg, &my_filter);
+  if (filter_id < 0) {
+    LOG_ERR("Unable to attach isr [%d]", filter_id);
+  }
+
+This example shows how to attach a callback from a work-queue.
```

Comment:
Is "This example" the code above or the code below?  I'd recommend having the description before the code.